### PR TITLE
openEO API v0.4.0

### PR DESCRIPTION
**Pull request by m-mohr**
**openEO API v0.4.0** (Release candidate, actual release is planned for 28/02/2019)

Currently, #158 may bring in some additional minor changes to the data discovery part. Also, processes may change slightly depending on feedback. Lastly, I still need to update several examples in the documentation. Nevertheless, **I'd highly appreciate if you could give it a final review before releasing it.**

* Human-readable version for easier review: https://open-eo.github.io/openeo-api/v/0.4.0/index.html
* Changelog: https://open-eo.github.io/openeo-api/v/0.4.0/changelog/

**Note**: Single source of truth are the actual sources for the rendered version (e.g. openapi.json, JSON files for processes, ...). For example, `cube:dimensions` in the collections is currently not shown in the rendered API documentation due to a bug in ReDoc, but it is still required to be implemented!

### CODE REVIEW

File: openapi.json [hunk 1]
```diff
@@ -1895,7 +2147,7 @@
         }
       },
       "post": {
-        "summary": "Publish a new secondary web service.",
+        "summary": "Publish a new service",
         "description": "Calling this endpoint will create a secondary web service such as WMTS, TMS or WCS. The underlying data is processes on-demand, but a process graph may simply access results from a batch job. Computations should be performed in the sense that it is only evaluated for the requested spatial / temporal extent and resolution.\n\n**Note:** Costs incured by shared secondary web services are usually paid by the owner, but this depends on the service type and whether it supports charging fees or not.",
```
Comment (tmistelbauer) [inline openapi.json]: Costs incured --> Costs incurred

```suggestion
        "description": "Calling this endpoint will create a secondary web service such as WMTS, TMS or WCS. The underlying data is processes on-demand, but a process graph may simply access results from a batch job. Computations should be performed in the sense that it is only evaluated for the requested spatial / temporal extent and resolution.\n\n**Note:** Costs incurred by shared secondary web services are usually paid by the owner, but this depends on the service type and whether it supports charging fees or not.",
```

Comment (tmistelbauer) [reply openapi.json]: I only found this minor issue (misspelling). Other than that, great job @m-mohr !

Comment (m-mohr) [reply openapi.json]: Thanks.

File: docs/collections.md [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
-STAC has several extensions (see their [repository](https://github.com/radiantearth/stac-spec)) that can be used to better describe your data. Clients and server are not required to implement all of them, so be aware that some clients may not be able to read all your meta data.
+STAC has several [extensions](https://github.com/radiantearth/stac-spec/tree/v0.6.2/extensions) that can be used to better describe your data. Clients and server are not required to implement all of them, so be aware that some clients may not be able to read all your meta data.
 
-Some commonly used extensions are:
+Some commonly used extensions that are integrated in the openEO API specification are:
 
-- EO extension (mostly integrated within openEO)
-- Scientific extension (integrated within openEO)
-- Dimensions extension (draft)
+- EO extension
```
Comment (mundialis-dev) [inline docs/collections.md]: Just a minor remark: is  "EO" vs "SAR" commonly distinguished? I'd rather contrast "Optical" vs "SAR" vs "Thermal" vs ...

Comment (m-mohr) [reply docs/collections.md]: Not sure about common sense, I'm just porting the STAC specification here. Unfortunately, STAC specifies EO as *Electro-Optical* and not *Earth Observation* in general, which is a bit confusing. I can rename the title for the schema to 'EO (Electro-Optical)' to make this clear.

Comment (mundialis-dev) [reply docs/collections.md]: Yes, that might reduce potential confusion, thanks.

Comment (m-mohr) [reply docs/collections.md]: Did so, thanks for spotting that.

**Comment by m-mohr** [comment]
The examples, especially https://open-eo.github.io/openeo-api/v/0.4.0/jobs/ needs to be updated.

**Comment by m-mohr** [comment]
**Last chance to review this and point out problems before the 0.4 release. All issues are solved now. I'll update the examples now and release tomorrow if nothing unexpected happens.**

**Pull request merged**